### ISSUE TITLE
Update Folder Crashes qTox on Startup

### ISSUE THREAD

**Issue by Mutus** [qTox/qTox#830]
Windows 7 issue. First time posting an issue on Github go easy on me. :)

I turn on tox today and it said it had an update that would be installed up restart, so i restarted and it just said the same thing. The build I had installed was  90e6ec6  I believe, well then i decided to download the lastest build  0fd3f5e installed it and then a nice fancy update box pops up and crashs instantly its at 5% on the bar. Restarted a few times and installed the older version again and it said it needed to do an updated on restart, then says update is corrupt when I restart. So I went into the tox config files folder on windows and saw a folder called "updates". I installed the lastest build again and started it up just to see, it crashed like it was. So then I deleted the updates folder (which I believe was from the older build) and qTox started right up.

May want figure out a way to handle that type of issue, the old build would not uses the update it downloaded and moving to the next build just causes that version to crash on startup cause the update file is not useable now. Not sure if this will be a continuing problem or not though. Just thought I'd share an issue I had.

**Comment by dubslow** [comment]
A commit was recently merged whose comment said it fixed a crash at 5%. Can you please update somehow (manually reinstall?) and let us know if this is still an issue?

**Comment by ghost** [comment]
@dubslow Dou you mean my PR https://github.com/tux3/qTox/pull/845? It didn't really fix any bug but gives out an error instead of just divide by zero and crash. Instead of  `fatalError(tr("The diff list is empty."));` we could delete the update folder and retry the update as it worked as you said.

What was the exact error message or did the program crash without saying anything? Have you MS VS or something similar with a debugger installed on your PC?

Has anyone else this issue?

**Comment by dubslow** [comment]
Does the fatal error prevent qTox from running after the updater fails?

**Comment by tux3** [comment]
This should be fixed, there was a problem with my Jenkins.

**Comment by Mutus** [comment]
Edit: OK so I guess the original problem is fixed, cause it goes farther then the original Issue I posted and it no longer crashes out, but still don't update the client. Here is my experience now though. Should I open a new Issue for this?

Ok so I downloaded "qTox Windows 2014-11-30 12:46:06" from the releases page, since I needed one that would update to a newer version. Client starts up (after I renabled check for update on startup) and window pops up saying "An update is available, etc." I click Yes. I have the /AppData/Roaming/Tox folder open and can see the Update folder get created and stuff getting downloaded into it. 

Btw some kind of progress bar here would be good visual feedback, and maybe the option to go ahead and restart once download is complete.

So after waiting 5 minutes to make sure stuff is downloaded all the way. I don't see files getting updated anymore in the folder after a few seconds anyway. I close the client out. On restart I get this box.

![capture2](https://cloud.githubusercontent.com/assets/9856558/5326148/391a8044-7cd5-11e4-9e9a-62cb8e6c254b.JPG)

Clicking Ok, deletes the Update folder. Client starts up and then up pops a "Update is availble window" all over again.

Do I need to try a version thats newer then "qTox Windows 2014-11-30 12:46:06" ? It was 6 days old and your message was 8 days old I figured that was good enough. The newest release I see right now is 4 days old.

**Comment by dubslow** [comment]
It very well could be related to the fact that qTox for Windows hasn't built successfully for a number of days (since the last available release, by definition), and it's possible that the update server is being updated with new version info, even though the corresponding binaries aren't actually available.

Let me get back to you on this one.

**Comment by cjpit** [comment]
I've never been able to get this to work on windows 8.1 :( it is always unable to update.

PS is there any reason that it downloads all of the files again? I've watched the update folder, and it copies all files/dlls/everything into the update folder.  qt files are massive etc (I'm on wireless, so always stingy with downloading).  I updated qtox in the last week by downloading the separate install from jenkins and doing it that way.  It seems the only way i can get it to update is doing it like that.  Is it possible it is just some problem with my env?
